lib/ansible/modules/cloud/cloudscale/cloudscale_server.py

```diff
@@ -0,0 +1,438 @@
+        if not 'uuid' in self.info:
+            return
+
+        # Can't use _get here because we want to handle 404
+        resp, info = fetch_url(self._module,
+                               API_URL+'servers/'+self.info['uuid'],
+                               headers=self._auth_header)
+        if info['status'] == 200:
+            self.info = self._transform_state(json.loads(resp.read()))
+        elif info['status'] == 404:
+            self.info = {'uuid': self.info['uuid'],
+                         'name': self.info.get('name', None),
+                         'state': 'absent'}
+        else:
+            self._module.fail_json(msg='Failure while calling the cloudscale.ch API for '
+                                       'update_info: %s' % info['body'])
+
+
+    def wait_for_state(self, state):
+        start = datetime.now()
+        while datetime.now() - start < timedelta(TIMEOUT_WAIT):
+            self.update_info()
+            if self.info['state'] == state:
+                return True
+            sleep(1)
+
+        return False
+
+
+    def create_server(self):
+        data = self._module.params.copy()
+
+        # check for required parameters to create a server
+        missing_parameters = []
+        for p in ('name', 'ssh_keys', 'image', 'flavor'):
+            if not p in data or not data[p]:
+                missing_parameters.append(p)
+
+        if len(missing_parameters) > 0:
+            self._module.fail_json(msg='Missing required parameter(s) to create a new server: %s' %
+                                   ' '.join(missing_parameters))
+
+        # Sanitize data dictionary
+        for k,v in data.items():
+
+            # Remove items not relevant to the create server call
+            if k in ('api_token', 'uuid', 'state'):
+                data.pop(k)
+                continue
+
+            # Remove None values, these don't get correctly translated by urlencode
+            if v is None:
+                data.pop(k)
+                continue
+
+        self.info = self._transform_state(self._post('servers', data))
+        self.wait_for_state('running')
+
+
+    def delete_server(self):
+        self._delete('servers/%s' % self.info['uuid'])
+        self.wait_for_state('absent')
+
+
+    def start_server(self):
+        self._post('servers/%s/start' % self.info['uuid'])
+        self.wait_for_state('running')
+
+
+    def stop_server(self):
+        self._post('servers/%s/stop' % self.info['uuid'])
+        self.wait_for_state('stopped')
+
+
+    def list_servers(self):
+        return self._get('servers')
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            state               = dict(default='running',
+                                       choices=['running',
+                                                'stopped',
+                                                'absent',
+                                       ]),
+            name                = dict(),
```

Comment:
I suppose ansible is internally using type='str', but I still think it would be better (more explicit) to mention it here. I feel like an empty dict is a bit weird.